### PR TITLE
chore: remove Docker image building machinery

### PR DESCRIPTION
**Pull request by SdgJlbl**
Fixes FL-1637

Companion PR
https://github.com/Substra/substra-tests/pull/364
https://github.com/owkin/substra-ci/pull/167

End-to-end tests https://github.com/owkin/substra-ci/pull/167#issuecomment-2230339043

### CODE REVIEW

File: Dockerfile
Comment (ThibaultFy) [inline Dockerfile:1]: We still use the dockerfile in substra-ci to build the substratools image https://github.com/owkin/substra-ci/blob/edb879efd3dca24aa84830a688c74c1e2dfc1eaa/ci/config.py#L187-L188 so we can't just delete it like this. 
But I think we can simplify it, and maybe match how its done in [substrafl](https://github.com/Substra/substrafl/tree/main/docker/substrafl-tests)

**Review by ThibaultFy — CHANGES_REQUESTED** [review]
DockerFile is used in substra-ci

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: Dockerfile
Comment (thbcmlowk) [reply Dockerfile:1]: Indeed, good catch @ThibaultFy! See e2e [workflow](https://github.com/owkin/substra-ci/actions/runs/9792458771/job/27038366634):
```
FATAL: One or more builds failed. See logs for more details
- [substra-tools]: https://console.cloud.google.com/cloud-build/builds;region=us-east1/766d2b38-5026-4ab5-9d79-bebeabec5e90?project=connect-314908
ERROR:root:docker image build(s) failed.
FATAL: docker image build(s) failed.
Traceback (most recent call last):
  File "/home/runner/work/substra-ci/substra-ci/./run-ci.py", line 302, in main
    build_images(config, KNOWN_HOST_FILE_PATH, RUN_TAG, DIR)
  File "/home/runner/work/substra-ci/substra-ci/ci/build_images.py", line 52, in build_images
    _wait_for_builds(cfg, tag, images, known_host_file_path, dir)
  File "/home/runner/work/substra-ci/substra-ci/ci/build_images.py", line 160, in _wait_for_builds
    raise Exception("docker image build(s) failed.")
Exception: docker image build(s) failed.
```

**Comment by Owlfred** [comment]
```
Usage: /e2e [options] [help]

/e2e may appear anywhere as long as it is on its own line

Options:
  --refs <value>                          Extra refs (branch or tag) with format REPO=GIT_REF,REPO=GIT_REF.
  Supported repositories: orchestrator, substra-backend, substra-frontend, substra-tools, substrafl, substra, substra-tests, substra-documentation, substra-ci.
  Example: /e2e --refs substra-backend=some_branch,orchestrator=some_tag (default: {})
  --tests-to-run, --tests <tests-to-run>  Comma-separated list of tests to run. Valid options: sdk, substrafl, doc, frontend, mnist, camelyon or NONE. Frontend tests will run automatically on substrafl, mnist and camelyon if selected. (default: "sdk")
  -h, --help                              display help for command
```

**Comment by SdgJlbl** [comment]
/e2e --refs substra-ci=chore/drop-substratools-image --tests sdk,substrafl

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by SdgJlbl** [comment]
/e2e --refs substra-ci=chore/drop-substratools-image --tests sdk,substrafl

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by SdgJlbl** [comment]
/e2e --refs substra-ci=chore/drop-substratools-image --tests sdk,substrafl

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: changes/101.removed [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove base Docker image.
```
Comment (guilhem-barthes) [inline changes/101.removed:1]: Should we indicate that the base image is replaced by python slim?

**Review by guilhem-barthes — APPROVED** [review]

**Pull request merged**